app/src/pages/authoring/[resourceType]/[id].tsx

```diff
@@ -86,18 +132,87 @@ export default function ResourceAuthoringPage() {
       <Grid>
         <Grid.Col span={6}>
           <Stack spacing="md">
-            <TextInput label="url" value={url} onChange={e => setUrl(e.target.value)} />
-            <TextInput label="identifier" value={identifier} onChange={e => setIdentifier(e.target.value)} />
+            <TextInput
+              label="url"
+              value={url}
+              onChange={e => setUrl(e.target.value)}
+              rightSection={
+                <ActionIcon
+                  onClick={() => {
+                    setUrl('');
+                  }}
+                >
+                  <X />
+                </ActionIcon>
+              }
+            />
+            <TextInput
+              label="identifier"
+              value={identifier}
+              onChange={e => setIdentifier(e.target.value)}
+              rightSection={
+                <ActionIcon
+                  onClick={() => {
+                    setIdentifier('');
+                  }}
+                >
+                  <X />
+                </ActionIcon>
+              }
+            />
+            <TextInput
+              label="name"
+              value={name}
+              onChange={e => setName(e.target.value)}
+              rightSection={
+                <ActionIcon
+                  onClick={() => {
+                    setName('');
+                  }}
+                >
+                  <X />
+                </ActionIcon>
+              }
+            />
+            <TextInput
+              label="title"
+              value={title}
+              onChange={e => setTitle(e.target.value)}
+              rightSection={
+                <ActionIcon
+                  onClick={() => {
+                    setTitle('');
+                  }}
+                >
+                  <X />
+                </ActionIcon>
+              }
+            />
+            <TextInput
+              label="description"
+              value={description}
+              onChange={e => setDescription(e.target.value)}
+              rightSection={
+                <ActionIcon
+                  onClick={() => {
+                    setDescription('');
+                  }}
+                >
+                  <X />
+                </ActionIcon>
+              }
+            />
             <Button
               w={120}
-              onClick={() =>
+              onClick={() => {
+                const [additions, deletions] = parseUpdate(url, identifier, name, title, description);
```

Comment:
Just a personal thought, but maybe you could disable the submit button until the user makes direct changes to one of the attributes in a TextInput?  This is more subjective so it really comes down to what you think looks the best :)

Reply:
I like this idea! I am going to see if there is a not-too-complicated way to do this.